src/tabs.ts

```diff
@@ -0,0 +1,307 @@
+import {LitElement, html} from 'lit';
+import {customElement, state, property} from 'lit/decorators.js';
+import {animate_defs, animationCSS, ResponsiveVal,
+        getResponsiveValues, parseAttrVal} from "./utils";
+import {styleMap} from 'lit/directives/style-map.js';
+
+@customElement('aalam-tabs')
+export class AalamTabs extends LitElement {
+    DEFAULT_VALUES:{[key:string]: any} = {
+        animationDur: 100,
+        activecls: 'tab-active',
+        fashion: 'xs:row',
+        colsize: "title:30%;body:70%",
+    }
+
+    @property({type:String, attribute:true})
+    animation = '';
+
+    @property({type:Number, attribute:true})
+    animationDur = this.DEFAULT_VALUES.animationDur;
+
+    @property({type:String, attribute:true})
+    activecls = this.DEFAULT_VALUES.activecls;
+
+    @property({type:String, attribute:true})
+    fashion = this.DEFAULT_VALUES.fashion;
+
+    @property({type:String, attribute:true})
+    colsize = this.DEFAULT_VALUES.colsize;
+
+    @state()
+    _internal_fashion:string = 'row';
+
+    private _cur_ix:number;
+    private _anim_number:string;
+    private _resizeListener = this._resizeEvent.bind(this);
+    private _node_insert_listener = this._nodeInserted.bind(this);
+    private _animation_styles:{[key:string]:string} = {};
+    private _column_size:{[key:string]:string} = {title:'30%', body:'70%'};
+    private _fashion_style:Array<ResponsiveVal> =
+                [{ll: 0, ul: null, val: "row", cond: "(min-width:0px)"}];
+
+    constructor() {
+        super();
+        this._anim_number = Math.random().toString(16).slice(2, 10);
+    }
+    override attributeChangedCallback(
+                name:string, old_val:string, new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if(name == 'animation') {
+            if(animate_defs[new_val])
+                this._animation_styles = {open: new_val, close: new_val}
+            else {
+                this._animation_styles = parseAttrVal(new_val, (v:string) => {
+                    try {
+                        if(Object.keys(animate_defs[v]))
+                            return true;
+                        else
+                            return false;
+                    } catch (err:any) {}
+                        return false;
+                });
+                this._animation_styles = Object.assign(
+                    {open:this._animation_styles?.show,
+                     close:this._animation_styles?.hide}, {});
```

Comment:
This is not what I intended. You can just assign the object to the animation_styles.